### PR TITLE
refactor: shared routing for E2E and CT

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes N/A

### Additional details

Basic refactor to reduce duplication. This also have benefits for the Test Replay feature, ensure all the CT dev server assets go via `networkProxy.handleHttpRequest` instead of `nodeProxy.web`. This allows us to correctly capture them and show the right assets during Test Replay for CT.

### Steps to test

- CI is green, so should be okay
- Tested using Test Replay. You can see our CT webpack tests in packages/reporter now look correct if you hit "Test Replay" https://cloud-staging.cypress.io/projects/9rttcd/runs/954/overview?roarHideRunsWithDiffGroupsAndTags=1

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: tooling/v8-snapshot/cache/darwin/snapshot-meta.json [hunk 1]
```diff
@@ -4073,13 +4073,11 @@
     "./packages/server/lib/project_utils.ts",
     "./packages/server/lib/remote_states.ts",
     "./packages/server/lib/request.js",
-    "./packages/server/lib/routes-ct.ts",
-    "./packages/server/lib/routes-e2e.ts",
     "./packages/server/lib/routes.ts",
     "./packages/server/lib/saved_state.ts",
     "./packages/server/lib/server-base.ts",
     "./packages/server/lib/server-ct.ts",
```
Comment (mike-plummer) [inline tooling/v8-snapshot/cache/darwin/snapshot-meta.json:4079]: Should `server-ct.ts` be gone too?

File: packages/server/lib/server-base.ts [hunk 2]
```diff
@@ -192,11 +221,90 @@ export abstract class ServerBase<TSocket extends SocketE2E | SocketCt> {
...
+              debug('ensuring baseUrl (%s) errored: %o', baseUrl, err)
+
+              return errors.get('CANNOT_CONNECT_BASE_URL_WARNING', baseUrl)
+            })
+          }
+        }).then((warning) => {
+          // once we open set the domain to root by default
+          // which prevents a situation where navigating
+          // to http sites redirects to /__/ cypress
+          this._remoteStates.set(baseUrl != null ? baseUrl : '<root>')
```
Comment (mike-plummer) [inline packages/server/lib/server-base.ts:301]: This is different behavior than exists in `server-ct` today - should we default to `<root>` regardless of testing type?

File: packages/server/lib/routes-ct.ts [hunk 3]
```diff
@@ -1,61 +0,0 @@
...
-  return send(req, pathToFile).pipe(res)
-}
-
-export const createRoutesCT = ({
-  config,
-  nodeProxy,
-}: InitializeRoutes) => {
-  const routesCt = Router()
-
-  routesCt.get(`/${config.namespace}/static/*`, (req, res) => {
```
Comment (mike-plummer) [inline packages/server/lib/routes-ct.ts:21]: I'm not seeing these routes in the consolidated `routes.ts` file - not necessary anymore?

**Review by mike-plummer — COMMENTED** [review]
Seeing some test failures so holding off on trying to validate locally until those are resolved. Main questions are around whether some of the consolidated e2e/ct behaviors from their respective `server-*` files should be guarded by testingType checks. Seems like CT tests are going to have extra logic around validating baseUrl accessibility, cookies, and domain injection that aren't there today - any risk/overhead for those being introduced?

File: packages/server/lib/routes-ct.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/server/lib/routes-ct.ts:21]: Seems we do not need these anymore - I guess they just get forwarded using the catch-all here https://github.com/cypress-io/cypress/pull/27618/files#diff-6370023360b8496cf29030186ae4c41f31dee48ea74c646f4153a8f0f9e368f2R218

It looks like some were not used already, so some of these routes are probably obsolete.

File: tooling/v8-snapshot/cache/darwin/snapshot-meta.json (hunk 1 above)
Comment (lmiller1990) [reply tooling/v8-snapshot/cache/darwin/snapshot-meta.json:4079]: Yes!

Comment (lmiller1990) [reply tooling/v8-snapshot/cache/darwin/snapshot-meta.json:4079]: I think these will be regenerated automatically anyway, but I'll remove `server-ct`.

File: packages/server/lib/server-base.ts (hunk 2 above)
Comment (lmiller1990) [reply packages/server/lib/server-base.ts:301]: Right I see, in `server-ct` we always set to `baseUrl`. I also noticed we have https://github.com/cypress-io/cypress/blob/9bd124e3089abeef818cd738fad46d10364eef12/packages/server/lib/server-base.ts#L493-L495

```js
    const baseUrl = this._baseUrl ?? '<root>'

    return this._remoteStates.set(baseUrl)
```

I think it makes no difference, ultimately we end up setting it to either `baseUrl` (always defined in CT) or `<root>`. If, for whatever reason `baseUrl` was not defined in CT, the `reset()` code I linked makes it `<root>` - so I think there's no actual change in behavior here.

**Review by ryanthemanuel — APPROVED** [review]
Looks good! Tests passing gives confidence on it for sure.

**Review by mike-plummer — APPROVED** [review]
LGTM, things seems to be working fine locally. Just want to verify this is intended to merge into `develop` vs release branch

**Comment by lmiller1990** [comment]
@mike-plummer develop is correct, this should be non breaking, although next release should be Cy 13 - either would be fine.

**Pull request merged**